website/project/views/node.py

```diff
@@ -915,6 +915,21 @@ def get_summary(auth, node, **kwargs):
         node, auth, primary=primary, link_id=link_id, show_path=show_path
     )
 
+@must_be_contributor_or_public
+def get_readable_descendants(auth, node, **kwargs):
+    descendants = []
+    for child in node.nodes:
+        if child.is_deleted:
+            continue
+        elif child.can_view(auth):
+            descendants.append(child)
+        else:
+            if not child.primary:
+                if node.has_permission(auth.user, 'read'):
```

Comment:
This should be `'write'`, because you need `write` access or higher to remove a `Private link`.

Nitpick: why not do `elif not child.primary and node.has_permission(auth.user, 'read'):`?